native/src/metadata/mod.rs

```diff
@@ -0,0 +1,33 @@
+use std::collections::HashMap;
+use crate::didcomm::Error;
+use crate::proto::okapi::okapi_metadata::{MetadataRequest, MetadataResponse};
+
+impl crate::Metadata {
+    pub fn get_metadata<'a>(request: &MetadataRequest) -> Result<MetadataResponse, Error<'a>> {
+        let env_vars = request.variables.clone();
+
+        // Get the version for sure, and parse it
+        let mut major_ver = 0;
+        let mut minor_ver = 0;
```

Comment:
Also include patch please. All are available as cargo vars. See link in my comment.